eppo_core/src/eval/eval_precomputed.rs

```diff
@@ -0,0 +1,326 @@
+                                action_categorical_attributes: selected_action.categorical.clone(),
+                            };
+
+                            [(precomputed_variation_value.clone(), precomputed_bandit)]
+                                .into_iter()
+                                .collect()
+                        } else {
+                            // If precomputed flag did not resolve to a value, we need to precompute all
+                            // bandits for the flag in case the user supplies a bandit variation as
+                            // default variation.
+                            configuration
+                                .flags
+                                .compiled
+                                .flag_to_bandit_associations
+                                .get(flag_key)?
+                                .iter()
+                                .filter_map(|(variation_value, bandit_variation)| {
+                                    let bandit_key = &bandit_variation.key;
+                                    let bandit_model = bandits.bandits.get(bandit_key)?;
+
+                                    let bandit_evaluation = bandit_model
+                                        .model_data
+                                        .evaluate(
+                                            flag_key,
+                                            subject_key,
+                                            subject_attributes,
+                                            actions,
+                                        )
+                                        .ok()?;
+
+                                    let selected_action = &actions[&bandit_evaluation.action_key];
+                                    let precomputed_bandit = PrecomputedBandit {
+                                        bandit_key: bandit_key.clone(),
+                                        action: bandit_evaluation.action_key,
+                                        action_probability: bandit_evaluation.action_weight,
+                                        optimality_gap: bandit_evaluation.optimality_gap,
+                                        model_version: bandit_model.model_version.clone(),
+                                        action_numeric_attributes: selected_action.numeric.clone(),
+                                        action_categorical_attributes: selected_action
+                                            .categorical
+                                            .clone(),
+                                    };
+
+                                    Some((variation_value.clone(), precomputed_bandit))
+                                })
+                                .collect()
+                        };
+
+                    Some((flag_key.clone(), flag_bandits))
+                })
+                .collect()
+        })
+        .unwrap_or_default();
```

Comment:
The diff in this file is a bit large because I renamed the file.

This highlighted piece is the main change, though

Reply:
Thanks for the note; yep I recognize the existing stuff 😄